polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1534,6 +1536,8 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
     }
 
     if (isa<clang::VectorType>(PTT) || isa<clang::ComplexType>(PTT)) {
+      if (auto VT = SubType.dyn_cast<mlir::VectorType>())
+        return mlir::MemRefType::get(1, SubType);
```

Comment:
See comment in the test case.

Reply:
Want to add a note here to indicate that the rank of this memref should be zero  (please reference the issue number you will open).

Reply:
https://github.com/intel/llvm/issues/7354